combat/CombatSystem.cpp

```diff
@@ -331,17 +331,11 @@ namespace {
 
                 boost::make_unique<Condition::And>(
                     boost::make_unique<Condition::Type>(OBJ_PLANET),
-                    boost::make_unique<Condition::Or>(
-                        boost::make_unique<Condition::Not>(
-                            boost::make_unique<Condition::MeterValue>(
-                                METER_DEFENSE,
-                                nullptr,
-                                boost::make_unique<ValueRef::Constant<double>>(0.0))),
-                        boost::make_unique<Condition::Not>(
-                            boost::make_unique<Condition::MeterValue>(
-                                METER_SHIELD,
-                                nullptr,
-                                boost::make_unique<ValueRef::Constant<double>>(0.0)))))));
+                    boost::make_unique<Condition::Not>(
+                        boost::make_unique<Condition::MeterValue>(
```

Comment:
Why remove the shield and defense tests?

Reply:
The implicit assumption is that you always have infrastructure if you have either defense or shields.

But that is maybe not necessarily so. So at least for robustness i will expand the condition